packages/data/src/test/factory.js

```diff
@@ -0,0 +1,135 @@
+/**
```

Comment:
Can you move this to a separate `registry-selectors.js` suite, just like #57510 does, to prevent unnecessary merge conflicts with that PR?

Reply:
Part of my rationale for creating a new file was that `test/factory.js` already has 2300 lines, it's inconvenient scrolling through that or trying to find something.